WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -435,10 +443,8 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
         return 2;
 
     } else if (sec == PostSettingsSectionMeta) {
-        if (self.apost.password) {
-            return 4;
-        }
-        return 3;
+        [self configureMetaSectionRows];
```

Comment:
Is there an advantage to putting this here in `numberOfRowsInSection` instead of in lifecycle method of the view controller itself? If (and I might be wrong) the number of cells is not changed after the view controller is initialized, it seems like this would be better of in an initializer or `viewDidAppear` or similar.

Reply:
Good thoughts, and I totally agree I'd love for this area to be as declarative as possible - it wasn't my first choice, but I ultimately decided to follow the pattern we did with `configureSections`.

The number of cells can change if the Visibility setting for the Post / Page changes (e.g. it will show or hide the Password row), so there is some dynamic mutating going on.

- I don't believe `viewDidAppear` has an effect because `viewWillAppear` already called `reloadData`, so we're too late.
- Putting it before `reloadData` in `viewWillAppear` semi-works, but the cell height isn't right.
- Putting it in the init probably won't work due to it needing to be dynamic.

I'll look into this some more and see if something better is available.

Reply:
> Putting it before reloadData in viewWillAppear semi-works, but the cell height isn't right.

Turns out this was a red herring that led me to the correct solution. I failed to do a proper lookup on the Password row in `heightForRowAtIndexPath`.

The proposed and implemented solution is:

- Calling `configureSections` in `viewWillAppear` right before we refresh the table with `reloadData`
- Properly detecting when we're calculating the height for the Password row